module/zfs/zfs_sa.c

```diff
@@ -231,6 +231,8 @@ zfs_sa_set_xattr(znode_t *zp)
 	error = nvlist_size(zp->z_xattr_cached, &size, NV_ENCODE_XDR);
 	if (error)
 		goto out;
+	if (size > SA_ATTR_MAX_LEN)
+		return (EFBIG);
```

Comment:
This function seems to make a point of avoiding early return. Perhaps this should be moved above the if (error) goto out; as if (error == 0 && size > SA_ATTR_MAX_LEN) error = EFBIG;